kroxylicious-kms-provider-aws-kms/src/main/java/io/kroxylicious/kms/provider/aws/kms/AwsKmsService.java

```diff
@@ -20,14 +21,28 @@
 @Plugin(configType = Config.class)
 public class AwsKmsService implements KmsService<Config, String, AwsKmsEdek> {
 
-    @NonNull
+    @SuppressWarnings("java:S3077") // Config is an immutable object
+    private volatile Config config;
+
     @Override
-    public AwsKms buildKms(Config options) {
-        return new AwsKms(options.endpointUrl(),
-                options.accessKey().getProvidedPassword(),
-                options.secretKey().getProvidedPassword(),
-                options.region(),
-                Duration.ofSeconds(20), options.sslContext());
+    public void initialize(@NonNull Config config) {
+        Objects.requireNonNull(config);
+        if (this.config != null) {
+            throw new IllegalStateException("KMS service is already initialized");
+        }
+        this.config = config;
     }
 
+    @NonNull
+    @Override
+    public AwsKms buildKms() {
+        if (config == null) {
+            throw new IllegalStateException("KMS service not initialized");
+        }
```

Comment:
This feels like we are being defensive against code we own. Its going to fail with an NPE anyway so I'm not sure the guard condition adds any real value.

Reply:
Removed.